sources/megaproxylist.go

```diff
@@ -0,0 +1,82 @@
+package sources
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/nfx/slrp/pmux"
+	"github.com/rs/zerolog/log"
+)
+
+func init() {
+	Sources = append(Sources, Source{
+		ID:        65,
+		Homepage:  "https://www.megaproxylist.net",
+		Frequency: 24 * time.Hour,
+		Seed:      true,
+		Feed:      simpleGen(Megaproxylist),
+	})
+}
+
+type megaproxylistLine struct {
+	address   string
+	port      string
+	country   string
+	reability string
+}
+
+var megaproxylistUrl = fmt.Sprintf("https://www.megaproxylist.net/download/megaproxylist-csv-%s_SDACH.zip", time.Now().Format("20060102"))
+
+// Scrapes https://www.megaproxylist.net
+func Megaproxylist(ctx context.Context, h *http.Client) (found []pmux.Proxy, err error) {
+	log.Info().Msg("Loading proxy checker database")
+
+	resp, err := http.Get(megaproxylistUrl)
+	if err != nil {
+        fmt.Println(err)
+		return
+	}
+	if resp.Body == nil {
+		return nil, fmt.Errorf("body is nil")
+	}
+	defer resp.Body.Close()
+	csv_data := unzipInMemory(ctx, resp)
+	r := csv.NewReader(bytes.NewBuffer(csv_data))
+	r.Comma = ';'
+	r.TrimLeadingSpace = true
+
+	// trick to skip header
+	if _, err := r.Read(); err != nil {
+		panic(err)
+	}
+
+	for {
+		record, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		/*
+			        if err != nil {
+						log.Fatal(err)
+					}
+		*/
+		fmt.Println(record)
+
+		proxy := megaproxylistLine{
```

Comment:
Why do you need an intermediate structure if you immediately convert it to pmux.Proxy?

Reply:
for some reason it doesn't want to go properly with record[0] and record[1], at least from my side script that i did for tests, i will retest